src/components/item/item-sliding.ts

```diff
@@ -47,19 +81,154 @@ import {List} from '../list/list';
  * @see {@link ../Item Item API Docs}
  * @see {@link ../../list/List List API Docs}
  */
+ // host: {
+ //   '[hidden]': '_hidden',
+ //   'class': 'toolbar',
+ //   '[class.statusbar-padding]': '_sbPadding'
+ // },
 @Component({
   selector: 'ion-item-sliding',
   template:
     '<ng-content select="ion-item,[ion-item]"></ng-content>' +
     '<ng-content select="ion-item-options"></ng-content>',
   changeDetection: ChangeDetectionStrategy.OnPush,
-  encapsulation: ViewEncapsulation.None,
+  encapsulation: ViewEncapsulation.None
 })
 export class ItemSliding {
+  private _openAmount: number = 0;
+  private _startX: number = 0;
+  private _offsetX: number = 0;
+  private _optsWidthRightSide: number = 0;
+  private _optsWidthLeftSide: number = 0;
+  private _sides: SideFlags;
+  private _timer: number = null;
+  private _leftOptions: ItemOptions;
+  private _rightOptions: ItemOptions;
+  private _optsDirty: boolean = true;
+
+  @ContentChild(Item) item;
 
-  constructor(@Optional() private _list: List, elementRef: ElementRef) {
+  constructor( @Optional() private _list: List, private _renderer: Renderer, private _elementRef: ElementRef) {
     _list.enableSlidingItems(true);
-    elementRef.nativeElement.$ionSlide = ++slideIds;
+    _elementRef.nativeElement.$ionComponent = this;
+  }
+
+  /**
+   * @private
+   */
+  @ContentChildren(ItemOptions)
+  set _itemOptions(itemOptions) {
+    var sides = 0;
+    for (var item of itemOptions.toArray()) {
+      var side = item.getSides();
+      switch(side) {
+        case SideFlags.Left: this._leftOptions = item; break;
+        case SideFlags.Right: this._rightOptions = item; break;
+      }
+      sides |= item.getSides();
+    }
+    this._optsDirty = true;
+    this._sides = sides;
+  }
+
+  /**
+   * @private
+   */
+  _startSliding(startX: number) {
+    this._addClass('active-slide');
+    this._offsetX = this._openAmount;
+    this._startX = startX;
+  }
+
+  /**
+   * @private
+   */
+  _moveSliding(x: number): number {
+    if(this._optsDirty) {
```

Comment:
I need to find a way to reconfigure my editor, I disable all the linting / formatting